### PR TITLE
Grant default organisation permissions

### PR DESCRIPTION
**Pull request by steventux**
## Context

Do not merge until we're ready to enable the org to org permissions feature.

We'd like to enable the organisation level permissions feature with minimal disruption to existing provider organisations.

## Changes proposed in this pull request

This migration:
- Grants `make_decisions` permission for all existing provider relationships and marks them as set up.
- Grants `manage_organisations` permission to all existing users with the manage_users permission.
- Grants `manage_organisations` permission to for orgs who don't have a managing user, this permission is granted to the user who agreed the DSA.
- Grants `manage_organisations` permission to the first user of any orgs who have an unsigned DSA.

## Guidance to review

- Should these updates be recorded as audit trail entries?
- The first statement to update `ProviderRelationshipPermissions` updates them into an invalid state because any 'set up' permissions should have enable a provider on each permission flag but we are only enabling `make_decisions`. Do we care about this? We're not forcing these providers to  fix this via the setup wizard. 

## Link to Trello card

https://trello.com/c/bQXtxL9Y/2682-data-migration-to-pre-set-all-org-org-permissions

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: db/migrate/20200922092516_add_default_permissions_for_provider_relationships.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+    # Grant manage organisations permissions for all provider users who can currently manage other provider users.
+    ProviderPermissions.where(manage_users: true, manage_organisations: false).update_all(manage_organisations: true)
+
+    # Find any remaining provider organisations without a user with manage_organisations permissions.
+    provider_ids_with_no_manage_org_permissions = ProviderPermissions.where(manage_organisations: false).pluck(:provider_id).uniq
+
+    if provider_ids_with_no_manage_org_permissions.any?
+      # Fetch users who accepted DSA for these unmanaged organisations.
+      provider_user_ids_from_agreements = ProviderAgreement.where(provider_id: provider_ids_with_no_manage_org_permissions).pluck(:provider_user_id).uniq
```
Comment (duncanjbrown) [inline db/migrate/20200922092516_add_default_permissions_for_provider_relationships.rb:18]: We should double check on prod that all these providers have such a user

**Review by duncanjbrown — APPROVED** [review]

File: db/migrate/20200922092516_add_default_permissions_for_provider_relationships.rb (hunk 1 above)
Comment (steventux) [reply db/migrate/20200922092516_add_default_permissions_for_provider_relationships.rb:18]: @duncanjbrown I can build this into the migration fairly easily.

Comment (steventux) [reply db/migrate/20200922092516_add_default_permissions_for_provider_relationships.rb:18]: On the other hand, so long as the provider permissions for the removed user were also removed, the update statement won't have any effect.

**Review by duncanjbrown — APPROVED** [review]
Tested locally — looks good to me!

**Pull request merged**